TrafficCapture/testingEnvironment/build.gradle

```diff
@@ -0,0 +1,125 @@
+        dependsOn ":${projectName}:jar"
+        from { project(":${projectName}").configurations.findByName("runtimeClasspath").files }
+        from { project(":${projectName}").tasks.getByName('jar') }
+        into artifactsDir
+        include "*.jar"
+        duplicatesStrategy = DuplicatesStrategy.WARN
+
+        if (projectName == "trafficCaptureProxyServerTest") {
+            include "*.properties"
+            from project.file("src/main/docker/${projectName}/jmeter.properties").absolutePath
+            into "${dockerBuildDir}"
+        }
+    }
+
+    task "createDockerfile_${projectName}"(type: com.bmuschko.gradle.docker.tasks.image.Dockerfile) {
+        dependsOn "copyArtifact_${projectName}"
+        destFile = project.file("${dockerBuildDir}/Dockerfile")
+        def baseImageOverrideProjectName = baseImageProjectOverrides.get(projectName)
+        if (baseImageOverrideProjectName) {
+            def dependentDockerImageName = dockerFilesForExternalServices.get(baseImageOverrideProjectName)
+            def hashNonce = calculateDockerHash(baseImageOverrideProjectName)
+            from "testenv/${dependentDockerImageName}:${hashNonce}"
+            dependsOn "buildDockerImage_${baseImageOverrideProjectName}"
+            runCommand("sed -i -e \"s|mirrorlist=|#mirrorlist=|g\" /etc/yum.repos.d/CentOS-* ;  sed -i -e \"s|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g\" /etc/yum.repos.d/CentOS-*")
+            runCommand("yum -y install nmap-ncat")
+        } else {
+            from 'openjdk:11-jre'
+            if (projectName == "trafficCaptureProxyServerTest") {
+                instruction 'COPY jmeter.properties /jmeter.properties'
+            }
+            runCommand("apt-get update && apt-get install -y netcat")
+        }
+
+        copyFile("jars", "/jars")
+        // can't set the environment variable from the runtimeClasspath because the Dockerfile is
+        // constructed in the configuration phase and the classpath won't be realized until the
+        // execution phase.  Therefore, we need to have docker run the command to resolve the classpath
+        // and it's simplest to pack that up into a helper script.
+        runCommand("printf \"#!/bin/sh\\njava -cp `echo /jars/*.jar | tr \\   :` \\\"\\\$@\\\" \" > /runJavaWithClasspath.sh");
+        runCommand("chmod +x /runJavaWithClasspath.sh")
+        // container stay-alive
+        defaultCommand('tail', '-f', '/dev/null')
+        //defaultCommand('/runJavaWithClasspath.sh', '...')
+    }
+}
+
+(javaContainerServices).forEach { projectName, dockerImageName ->
+    def dockerBuildDir = "build/docker/${projectName}"
+    task "buildDockerImage_${projectName}"(type: DockerBuildImage) {
+        dependsOn "createDockerfile_${projectName}"
+        inputDir = project.file("${dockerBuildDir}")
+        images.add("testenv/${dockerImageName}:${version}")
+        images.add("testenv/${dockerImageName}:latest")
+    }
```

Comment:
This should be refactored into a common shared class and imported by both projects